ets.md

```diff
@@ -0,0 +1,926 @@
+* Възможно е да променим собственика на таблица и ръчно, даже, когато процесът-собственик на таблицата още съществува.
+
+---
+### Наследници
+* Това става с :ets.give_away/3 (:ets.give_away(table, heir_pid, transfer_data)).
+* По подразбиране наследника има стойност :none. Винаги можем да премахнем наследник, като зададем тази стойност.
+
+---
+### Компресия
+
+* Можем да зададем опцията :compressed на таблица.
+* Това означава, че всичко освен ключа на даден запис ще е компресирано.
+* Това ще оптимизира паметта използвана от таблицата.
+* Цената е, че четене на цели редове ще стане по-бавно.
+* По бавно ще стане и добавянето на нови записи.
+
+---
+### Опции четене и писане
+
+* Има две опции за оптимизиране на често среащни случаи.
+* И двете по подразбиране са изключени (false).
+
+---
+### Опции четене и писане
+
+* read_concurrency: true - Ако имаме множество четения без писания често.
+* Ако го направим и често редуваме четения и писания, достъпът до таблицата ще се влоши.
+
+---
+### Опции четене и писане
+
+* write_concurrency: true - Ако имаме много промени.
+* Промени от типа на insert, delete.
+* Ако редуваме писането с чести четения, това не е добра идея.
+* По-бързите конкурентни писания идват с цената на повече използвана памет.
+
+---
+### Опции четене и писане
+
+* Можем да комбинираме двете опции ако имаме периоди на много четения и периоди на множество писания.
+* Тези опции са свързани с начина на заключване и синхронизиране на достъпа до записите в таблиците.
+* Най-добре е с наблюдение и тестване да определите стойностите на тези опции.
+
+---
+### Кога да използваме ETS?
+
+---
+![Image-Absolute](assets/when.jpg)
+
+---
+### Кога да използваме ETS?
+* Споделено състояние между процеси, които да имат конкурентен и бърз достъп до него.
+* Пазене на текущото състояние на процес, което да бъде възстановено, ако процеса умре.
+* Вместо Map или List за оптимизация, макар че не винаги това ще е най-бързото решение.
+* За регистриране на познати процеси.
+
+---
+### Кога да НЕ използваме ETS?
+* Когато искаме да работим с данни, които могат да бъдат изпращани от един node към друг.
+* Когато искаме дистрибутирана база данни върху множество node-ове.
+* Когато искаме да запазим данните си между рестартирания на node-а си.
+
+---
+### Кога да НЕ използваме ETS?
+* В повечето случаи проста комуникация между процеси върши работа.
+* При тестване, ако имаме нужда от временно състояние в паметта по-добре да ползваме Agent.
+
+---
+### Кога да НЕ използваме ETS?
+* При мемоизация на някакъв резултат, докато си строим данни, които ще използваме по-късно в даден процес, по-добре да ползваме структура в heap-а на този процес.
+* ETS има основната задача да пребори single-process bottleneck проблема. Ако данните ни живеят само в процесът, който ги ползва или ако само един процес ги чете/пише от текущия, ETS не е необходима.
+
+---
+### Абстракции над ETS : DETS
```

Comment:
Някакъв use-case да имаш? Аз не съм го ползвал досега.
Единственото срещане на `dets` в кода и зависимостите ми е в  [cachex](https://github.com/whitfin/cachex), но срещането е да кажат, че не го ползват:
```
# Rather than using DETS to back up the internal ETS table, this module will
# serialize the entire table using the ETF via the `Cachex.Disk` module.
```